src/metrics/misc.ts

```diff
@@ -17,3 +26,163 @@ export function countGamesInDataset(datasetPath: string): number {
+  }
+
+  // Aggregate the results of the metric
+  aggregate() {
+    // Calculate the average player rating after each game
+    this.averagePlayerRating = this.totalPlayerRating / (this.numberGamesAnalyzed * 2);
+
+    // Calculate the average player rating diff
+    this.averageRatingDiff = this.totalPlayerRatingDiff / this.numberGamesAnalyzed
+
+    // Calculate the player with the most games played
+    let maxGames = 0;
+    let playerMostGames = '';
+    for (const player in this.playerGameStats) {
+      if (this.playerGameStats[player] > maxGames) {
+        maxGames = this.playerGameStats[player];
+        playerMostGames = player;
+      }
+    }
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+    };
+  }
+
+  // Reset the maps used to track metrics
+  clear(): void {
+    this.numberGamesAnalyzed = 0;
+    this.averagePlayerRating = 0;
+    this.averageRatingDiff = 0;
+    this.largestRatingDiff = 0;
+    this.largestRatingDiffGame = '';
+    this.mostGamesPlayed = 0;
+    this.playerMostGames = '';
+    this.gameTypeStats = {
+      numberBulletGames: 0,
+      numberBlitzGames: 0,
+      numberRapidGames: 0,
+      numberClassicalGames: 0,
+    };
+    this.gameTimeControlStats = {};
+
+    this.totalPlayerRating = 0;
+    this.totalPlayerRatingDiff = 0;
+    this.playerGameStats = {};
+
+  }
+
+  processGame(
+    game: { move: PrettyMove; board: Piece[] }[],
+    metadata?: string[]
+  ) {
+
+    // Update the gameTimeControlStats based on the time control of the game
+    const timeControl = metadata?.find((data) => data.startsWith('[TimeControl'));
+    if (timeControl) {
+      const timeControlQuantity = timeControl.split(' ')[1];
+      if (this.gameTimeControlStats[timeControlQuantity]) {
+        this.gameTimeControlStats[timeControlQuantity]++;
+      } else {
+        this.gameTimeControlStats[timeControlQuantity] = 1;
+      }
+    }
+    
+    // Identify the time control type from the metadata
+    const gameType = metadata?.find((data) => data.startsWith('[Event'));
+    if (gameType) {
+      if (gameType.includes('Bullet')) {
+        this.gameTypeStats.numberBulletGames++;
+      } else if (gameType.includes('Blitz')) {
+        this.gameTypeStats.numberBlitzGames++;
+      } else if (gameType.includes('Rapid')) {
+        this.gameTypeStats.numberRapidGames++;
+      } else if (gameType.includes('Classical')) {
+        this.gameTypeStats.numberClassicalGames++;
+      }
+    }
+
+    // Calculate average player rating (first, find the totalPlayerRating)
+    const whiteRating = parseInt(metadata?.find((data) => data.startsWith('[WhiteElo'))?.replace(/"/g, '').split(' ')[1]);
+    const blackRating = parseInt(metadata?.find((data) => data.startsWith('[BlackElo'))?.replace(/"/g, '').split(' ')[1]);
```

Comment:
If the format is going to be super consistent on these a substring will be better

Reply:
gpt4 says:

Yes, you can use the substring method to extract the ratings. However, you need to know the exact position where the rating starts and ends. If the position is not fixed, it's better to use split as it is more flexible.

Here's how you can do it with substring if the position of the rating is always fixed:
const whiteRatingStr = metadata?.find((data) => data.startsWith('[WhiteElo'));
const blackRatingStr = metadata?.find((data) => data.startsWith('[BlackElo'));

const whiteRating = parseInt(whiteRatingStr?.substring(startPos, endPos).replace(/"/g, ''));
const blackRating = parseInt(blackRatingStr?.substring(startPos, endPos).replace(/"/g, ''));

Replace startPos and endPos with the actual positions.

**Please note that this approach is less flexible and more error-prone than using split if the position of the rating is not always the same.**

Reply:
"If the format is going to be super consistent"
And don't just take GPTs word.
If we know the index of the elo metadata we don't need to find it, we can just index & substr. Compiling regex adds some complexity over time